source/edge-server/permissions.txt

```diff
@@ -0,0 +1,274 @@
+documents using the ``user_id``. 
+
+For example, if an Edge Server instance represents a healthcare clinic, you 
+might only want to sync data relevant to that clinic's patients.
+
+In this example, every document has a ``facility_id`` property whose value is
+the Edge Server's ``user_id``. This role means that the Edge Server can only
+read and write documents with a ``facility_id`` matching its own ``user_id``.
+
+.. code-block:: json
+
+   {
+     "name": "facilityItemsOnly",
+     "apply_when": {},
+     "document_filters": {
+       "write": {
+         "facility_id": "%%user.id"
+       },
+       "read": {
+         "facility_id": "%%user.id"
+       },
+     },
+     "read": true,
+     "write": true,
+     "insert": true,
+     "delete": true,
+     "search": true
+   }
+
+Read All, Write Own Documents
++++++++++++++++++++
+
+You can configure an Edge Server instance to be able to read all data, but
+write only its own data. For example, in a retail setting, an Edge Server 
+instance might represent a store inventory system. You might want the store to
+have the ability to search across all inventory for an item, but only sell 
+items in its own inventory.
+
+.. code-block:: json
+   :emphasize-lines: 5-8
+
+   {
+     "name": "readAllWriteOnlyStoreItems",
+     "apply_when": {},
+     "document_filters": {
+       "write": {
+         "store_id": "%%user.id"
+       },
+       "read": true
+     },
+     "read": true,
+     "write": true,
+     "insert": true,
+     "delete": true,
+     "search": true
+   }
+
+Client Permissions
+~~~~~~~~~~~~~~~~~~
+
+Clients that connect to an Edge Server instance have their own permissions. 
+You can think of an Edge Server instance's permissions as a filter between the
+data in Atlas and the data that a client can read and write.
+
+In a healthcare setting similar to the Edge server instance example above, the
+Edge Server instance may only read and write data that is relevant to its own 
+facility. A client may then further restrict the permissions. For example, a 
+patient in the facility may only view their own data.
+
+You represent these as independent roles within the ``roles`` array.
+
+In this example, a ``PatientRecords`` database has a ``Visits`` collection.
+The two roles determine which documents the Edge Server instance can sync,
+and which documents an individual client connected to the instance can sync:
+
+- The ``facilityItemsOnly`` role filters items that sync to the Edge Server
+  instance. The only items that sync to the facility are records where the
+  ``facility_id`` matches the Edge Server instance ``user_id``.
+- The ``patientOwnRecordsOnly`` role filters items that can sync to a connected
+  client device. The only items that sync to a connected client are visits 
+  where the ``patient_id`` is the client's ``user_id``.
+
+.. code-block:: json
+
+   {
+     "collection": "Visits",
+     "database": "PatientRecords",
+     "roles": [
```

Comment:
Do you think it is worth calling out here that it is important that the `facilityItemsOnly` role is there before the `patientOwnRecordsOnly` role?

Reply:
Yes, I meant to mention this but totally forgot. Thanks for calling it out!